File: SkillSync.Client/src/app/shared/entities/models/feature.ts
import { FeatureInputType } from "../../enums/feature-input-type";
import { DropdownOption } from "./dropdown-option";

export interface Feature {
    id: string,
    name: string,
    inputType: FeatureInputType,
    dropdownOptionId?: string
    dropdownOption?: DropdownOption,
}

File: SkillSync.WebServer/SkillSync.Domain/Models/Feature.cs
using SkillSync.Domain.Enums;

namespace SkillSync.Domain.Models
{
    public class Feature
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public FeatureInputType InputType { get; set; }

        public Guid? DropdownOptionId { get; set; }
        public DropdownOption? DropdownOption { get; set; }
    }
}